ITR/data/base_providers.py

```diff
@@ -1,25 +1,27 @@
 import warnings  # needed until quantile behaves better with Pint quantities in arrays
 import numpy as np
 import pandas as pd
+import pint
+
 from functools import reduce, partial
 from operator import add
 from typing import List, Type, Dict
-import logging
 
-from ITR.data.osc_units import Q_, PA_
-from ITR.configs import ColumnsConfig, TemperatureScoreConfig, VariablesConfig, LoggingConfig
+import ITR
```

Comment:
i understand that you don't see `import ITR` as a big deal, but i see it as mix of dependencies, which may lead to a poorer SW architecture, and eventually, to lower quality of the SW.
we already use the pattern of `from ITR.<...> import <...>`, as we see in lines below.
i suggest that we continue following the pattern

Reply:
Note that
```
import ITR
```
is different than
```
from ITR import *
```
The former allows us to later use the ITR prefix when referencing bits we know we want to use whereas the latter just fills our namespace with everything defined in ITR.

The
```
from ITR import xyz
```
puts xyz into our namespace, which in some cases we want.  But if we want to import something and highlight that it came from ITR (such as ITR.isnan) we cannot do other than importing ITR to get the ITR handle.  At least that's how I understand it.

Reply:
The is_nan function is intended to be polymorphic between the NumPy numeric types and uncertainties.  The former are understood by Pandas as part of the Dtype hierarchy.  The latter use wrappers that work outside the ExtensionArray context.  Creating a joint class of those two things would be a nightmare.  I think it's fine to say that ITR, as a module, will define polymorphic functions as needed to bridge those worlds.  If you have an OOP idea that's better, I'm all ears.

Reply:
yes, i see, moving ‘is_nan‘ to some "aggregate" class is no path.
let's see what else we have.
we could resolve my problem with `import ITR`, by moving the `HAS_UNCERTAINTIES` import logic into `__init__.py` of some other package.
for example, if we moved content of `ITR/__init__.py` into, for example, a separate package `ITR/uncert/__init__.py` - then we could do `from ITR import uncert`.
It would give us the desired result, and keep clean dependencies at the same time